src/StatefulButton/index.jsx

```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import Icon from '../Icon';
-import Button from '../Button';
+import { Button } from '../';
```

Comment:
Hmm, how does this impact tree-shaking?  Since this is pulling from the top-level index.js file, won't that cause all the imports from that file to get included in the bundle?

Reply:
I don't think this is the case. The es6 exports from that index.js are atomic no? Importing it here is the same as importing Button in a project (right?).

Reply:
That wasn't my understanding... like, I thought it basically imports the tree from whatever entry point you choose.  So if you import from Button itself, it'll only get it's dependencies, but if you import from the top-level index I thought you'd get everything that file depends on... maybe I'm mis-remembering?

Reply:
We discussed over slack, but for posterity we think we're good here: https://webpack.js.org/guides/tree-shaking/ 

```
So, what we've learned is that in order to take advantage of tree shaking, you must...

- Use ES2015 module syntax (i.e. import and export).
- Ensure no compilers transform your ES2015 module syntax into CommonJS modules (this is the default behavior of the popular Babel preset @babel/preset-env - see the documentation for more details).
- Add a "sideEffects" property to your project's package.json file.
- Use the production mode configuration option to enable various optimizations including minification and tree shaking.